Lib/tty.py

```diff
@@ -15,22 +15,60 @@
 OSPEED = 5
 CC = 6
 
-def setraw(fd, when=TCSAFLUSH):
-    """Put terminal into a raw mode."""
-    mode = tcgetattr(fd)
-    mode[IFLAG] = mode[IFLAG] & ~(BRKINT | ICRNL | INPCK | ISTRIP | IXON)
-    mode[OFLAG] = mode[OFLAG] & ~(OPOST)
-    mode[CFLAG] = mode[CFLAG] & ~(CSIZE | PARENB)
-    mode[CFLAG] = mode[CFLAG] | CS8
-    mode[LFLAG] = mode[LFLAG] & ~(ECHO | ICANON | IEXTEN | ISIG)
+def cfmakeraw(mode):
```

Comment:
The original set of flags seems a bit arbitrary. Although I have marked the flags as `POSIX.1-2017`, they will also work on old systems since the same set of flags is present in The Open Group Base Specifications Issue 6 (2004 edition): https://pubs.opengroup.org/onlinepubs/009695399/basedefs/xbd_chap11.html. This is not surprising at all since this is not new technology.

Also, if you wish, you can take a look at what `raw` means to `stty`: https://pubs.opengroup.org/onlinepubs/9699919799/utilities/stty.html. We could add other functions that mimic various functions of `stty`, such as ones that put terminals in `cooked mode`, `sane mode` etc.